hls4ml/templates/vivado/nnet_utils/nnet_common.h

```diff
@@ -32,6 +32,9 @@ namespace nnet {
 enum io_type {io_parallel = 0, io_serial, io_stream};
 enum strategy { latency, resource };
 
+// Activation enum
+enum activ_type {activ_relu = 0, activ_sigmoid, activ_tanh, activ_softmax};
+
```

Comment:
I think we should find a nicer mechanism to enable multiple activation functions from the RNN. One method could be to make classes for the activation functions, and then include in the RNN layer config the object that corresponds to the appropriate activation. 

The aim would be to eliminate the `if (CONFIG_T::activation_type == activation_relu) { relu<...>(...) }` constructions in the HLS and replace it with a single line like:
`CONFIG_T::template activation<...>::activation(...)`

It's similar to what we do now with the `nnet::mult`  - classes [here](https://github.com/fastmachinelearning/hls4ml/blob/master/hls4ml/templates/vivado/nnet_utils/nnet_mult.h), config [here](https://github.com/fastmachinelearning/hls4ml/blob/master/hls4ml/templates/vivado/nnet_utils/nnet_dense.h#L33-L34), and call to product [here](https://github.com/fastmachinelearning/hls4ml/blob/master/hls4ml/templates/vivado/nnet_utils/nnet_dense_latency.h#L96).

Reply:
Followed the same method used for `nnet::mult` and passing objects for appropriate activation now.